src/GraphQL.Dynamic/LiteralGraphType/LiteralGraphTypeHelpers.cs

```diff
@@ -0,0 +1,141 @@
+using System;
+using System.Collections.Concurrent;
+using System.Linq;
+using GraphQL.Resolvers;
+using GraphQL.Types;
+
+namespace GraphQL.Dynamic.Types.LiteralGraphType
+{
+    internal delegate FieldType FieldTypeResolver(LiteralGraphTypeMemberInfo m);
+    internal delegate object GetValueFn(ResolveFieldContext context);
+
+    internal static class LiteralGraphTypeHelpers
+    {
+        private static ConcurrentDictionary<string, FieldTypeResolver> MemberFieldTypeResolverMappings { get; set; } = new ConcurrentDictionary<string, FieldTypeResolver>();
+        private static ConcurrentDictionary<string, FieldType> MemberFieldTypeMappings { get; set; } = new ConcurrentDictionary<string, FieldType>();
+
+        private static ConcurrentDictionary<LiteralGraphTypeMemberInfoType, Type> LiteralGraphTypeMemberInfoTypeMappings { get; set; } = new ConcurrentDictionary<LiteralGraphTypeMemberInfoType, Type>();
+        private static ConcurrentDictionary<LiteralGraphTypeMemberInfoType, FieldTypeResolver> LiteralGraphTypeMemberInfoFieldTypeResolverMappings { get; set; } = new ConcurrentDictionary<LiteralGraphTypeMemberInfoType, FieldTypeResolver>();
+
+        static LiteralGraphTypeHelpers()
+        {
+            PopulateMappings();
+        }
+
+        private static void PopulateTypeMapping(LiteralGraphTypeMemberInfoType type, Type actualType, FieldTypeResolver resolver)
+        {
+            LiteralGraphTypeMemberInfoTypeMappings.TryAdd(type, actualType);
+            LiteralGraphTypeMemberInfoFieldTypeResolverMappings.TryAdd(type, resolver);
+        }
+
+        private static void PopulateMappings()
+        {
+            PopulateTypeMapping(LiteralGraphTypeMemberInfoType.String, typeof(string), CreateFieldType<StringGraphType>);
+            PopulateTypeMapping(LiteralGraphTypeMemberInfoType.Int, typeof(int), CreateFieldType<IntGraphType>);
+            PopulateTypeMapping(LiteralGraphTypeMemberInfoType.Long, typeof(long), CreateFieldType<IntGraphType>);
+            PopulateTypeMapping(LiteralGraphTypeMemberInfoType.Float, typeof(float), CreateFieldType<FloatGraphType>);
+            PopulateTypeMapping(LiteralGraphTypeMemberInfoType.Double, typeof(double), CreateFieldType<FloatGraphType>);
+            PopulateTypeMapping(LiteralGraphTypeMemberInfoType.Boolean, typeof(bool), CreateFieldType<BooleanGraphType>);
+            PopulateTypeMapping(LiteralGraphTypeMemberInfoType.Guid, typeof(Guid), CreateFieldType<IdGraphType>);
+            PopulateTypeMapping(LiteralGraphTypeMemberInfoType.DateTimeOffset, typeof(DateTimeOffset), CreateFieldType<DateGraphType>);
```

Comment:
DateTime is missing:

```
PopulateTypeMapping(LiteralGraphTypeMemberInfoType.DateTime, typeof(bool), CreateFieldType<DateGraphType>);
```

Reply:
Also: 
I'm not so sure about this line 
```csharp
PopulateTypeMapping(LiteralGraphTypeMemberInfoType.Guid, typeof(Guid), CreateFieldType<IdGraphType>);
```
because a GUID is not certainly a `IdGraphType` as a `IdGraphType` is not always a GUID
https://github.com/graphql-dotnet/graphql-dotnet/blob/c4bc7ca006d3ab5d7e60089c12e8e7038ee3797f/src/GraphQL/Types/IdGraphType.cs#L27

Reply:
@OneCyrus: fixed as part of my merges; thanks for the great find 😄 
@PascalSenn: yeah, honestly I don't know the GraphQL spec well enough to know the right way to handle this.  I just know that it's supposed to be a unique value that serializes as a string -- any help would be appreciated here!